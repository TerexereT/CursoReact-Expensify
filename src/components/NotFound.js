import React from 'react'
import {Link} from 'react-router-dom'

const NotFoundDBPage = () => (
    <div>
        error 404! Page Not Found
        - <Link to="/">Home</Link>
    </div>
)

export default NotFoundDBPage