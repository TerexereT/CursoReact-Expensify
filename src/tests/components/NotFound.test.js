import React from 'react'
import NotFound from '../../components/NotFound'
import { shallow } from 'enzyme'

test('NotFound render', () => {
    const wrapper = shallow(<NotFound />)
    expect(wrapper).toMatchSnapshot()
})