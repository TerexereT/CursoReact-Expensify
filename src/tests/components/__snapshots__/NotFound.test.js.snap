// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFoundDBPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "error 404! Page Not Found - ",
        <Link
          to="/"
>
          Home
</Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "error 404! Page Not Found - ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Home",
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "error 404! Page Not Found - ",
          <Link
            to="/"
>
            Home
</Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "error 404! Page Not Found - ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Home",
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
